src/examples/euler/EulerFullRangeHook.sol

```diff
@@ -0,0 +1,112 @@
+    function _recipient() internal view override returns (address) {
+        return swapAccount;
+    }
+
+    /// @dev computes the tick range of the JIT position by returning ticks as +/- 10% of spot price
+    /// @inheritdoc JIT
+    function _getTickRange(PoolKey memory poolKey, uint128, /*amount0*/ uint128, /*amount1*/ uint160 sqrtPriceX96)
+        internal
+        pure
+        override
+        returns (int24 tickLower, int24 tickUpper)
+    {
+        // calculating sqrt(price * 0.9e18/1e18) * Q96 is the same as
+        // (sqrt(price) * Q96) * (sqrt(0.9e18/1e18))
+        // (sqrt(price) * Q96) * (sqrt(0.9e18) / sqrt(1e18))
+        uint160 _sqrtPriceLower = uint160(
+            FixedPointMathLib.mulDivDown(
+                uint256(sqrtPriceX96), FixedPointMathLib.sqrt(0.9e18), FixedPointMathLib.sqrt(1e18)
+            )
+        );
+
+        // calculating sqrt(price * 1.1) * Q96 is the same as
+        // (sqrt(price) * Q96) * (sqrt(1.1e18/1e18))
+        // (sqrt(price) * Q96) * (sqrt(1.1e18) / sqrt(1e18))
+        uint160 _sqrtPriceUpper = uint160(
+            FixedPointMathLib.mulDivDown(
+                uint256(sqrtPriceX96), FixedPointMathLib.sqrt(1.1e18), FixedPointMathLib.sqrt(1e18)
+            )
+        );
+
+        int24 _tickLowerUnaligned = TickMath.getTickAtSqrtPrice(_sqrtPriceLower);
+        int24 _tickUpperUnaligned = TickMath.getTickAtSqrtPrice(_sqrtPriceUpper);
+
+        // align the ticks to the tick spacing
+        int24 tickSpacing = poolKey.tickSpacing;
+        tickLower = _alignTick(_tickLowerUnaligned, tickSpacing);
+        tickUpper = _alignTick(_tickUpperUnaligned, tickSpacing);
+    }
+
+    // Utility Functions
+
+    function _sendToPoolManager(address vault, Currency currency, uint128 amount) private returns (uint128) {
+        uint256 balance = _myBalance(vault);
+
+        uint256 available;
+        if (balance > 0) {
+            poolManager.sync(currency);
+
+            available = amount < balance ? amount : balance;
+            _withdrawFromEuler(vault, swapAccount, swapAccount, address(poolManager), available);
+
+            poolManager.settle();
+        }
+
+        return uint128(available);
+    }
+
+    /// @dev NOT PRODUCTION READY. Incorrectly rounds ticks to 0
+    /// Should be rounding consistently either towards spot price or away from spot price, regardless of sign
+    function _alignTick(int24 tick, int24 tickSpacing) internal pure returns (int24) {
+        return (tick / tickSpacing) * tickSpacing;
+    }
+
+    function _getInputOutputAndAmount(PoolKey calldata key, IPoolManager.SwapParams calldata params)
+        internal
+        pure
+        returns (Currency input, Currency output, uint256 amount)
+    {
+        (input, output) = params.zeroForOne ? (key.currency0, key.currency1) : (key.currency1, key.currency0);
+
+        amount = params.amountSpecified < 0 ? uint256(-params.amountSpecified) : uint256(params.amountSpecified);
```

Comment:
super clean function that deserves to be in a library in `utils/SwapParser.sol` maybe?

I like library directives for the syntactic sugar

```solidity
(Currency input, Currency output, bool exactInput, uint256 amount) = key.getInputOutput(params)
```

Reply:
Yeah that looks nice! can make a separate PR for it.